src/plugins/timeseries-plugin.ts

```diff
@@ -0,0 +1,258 @@
+import { OffscreenRenderEngine } from '../engines/offscreen-render-engine';
+import { SeparatedInteractionsEngine } from '../engines/separated-interactions-engine';
+import { HitRegion, RegionTypes } from '../types';
+import UIPlugin from './ui-plugin';
+
+export type TimeseriesPoint = [number, number];
+
+interface TimeseriesPointsSummary {
+    min: number;
+    max: number;
+    first: number;
+    last: number;
+}
+
+export type TimeseriesPluginStyles = {
+    height?: number;
+    padding?: number;
+    color?: string;
+};
+
+type TimeseriesPluginStylesNoOptional = {
+    height: number;
+    padding: number;
+    color: string;
+};
+
+export const defaultCPUPluginStyles: TimeseriesPluginStylesNoOptional = {
+    height: 68,
+    padding: 5,
+    color: 'red',
+};
+
+type TimeseriesPointWithIndex = [idx: number, ts: number, value: number, normalizedTs: number, normalizedValue: number];
+
+enum TimeseriesPointWithIndexX {
+    index = 0,
+    timestamp = 1,
+    value = 2,
+    normalizedTs = 3,
+    normalizedValue = 4,
+}
+
+export class TimeseriesPlugin extends UIPlugin<TimeseriesPluginStylesNoOptional> {
+    height: number;
+    data: TimeseriesPoint[];
+    maxValue: number;
+    hoveredRegion: HitRegion<{}> | null = null;
+    selectedRegion: HitRegion<{}> | null = null;
+    summary: TimeseriesPointsSummary | null = null;
+    override styles: TimeseriesPluginStylesNoOptional;
+
+    constructor({ name, data, styles }: { name: string; data: TimeseriesPoint[]; styles?: TimeseriesPluginStyles }) {
+        super(name);
+
+        this.styles = { ...defaultCPUPluginStyles, ...(styles ?? {}) };
+        this.data = [];
+        this.name = name;
+
+        this.height = this.styles.height;
+
+        this.setData(data);
+    }
+
+    override init(renderEngine: OffscreenRenderEngine, interactionsEngine: SeparatedInteractionsEngine) {
+        super.init(renderEngine, interactionsEngine);
+
+        this.interactionsEngine.on('change-position', this.handlePositionChange.bind(this));
+        this.interactionsEngine.on('hover', this.handleHover.bind(this));
+
+        this.interactionsEngine.on('up', this.handleMouseUp.bind(this));
+    }
+
+    handlePositionChange(position: { deltaX: number }) {
+        const startPositionX = this.renderEngine.parent.positionX;
+
+        this.interactionsEngine.setCursor('grabbing');
+
+        this.renderEngine.tryToChangePosition(position.deltaX);
+
+        if (startPositionX !== this.renderEngine.parent.positionX) {
+            this.renderEngine.parent.render();
+        }
+    }
+
+    handleMouseUp() {
+        this.interactionsEngine.clearCursor();
+    }
+
+    setData(data: TimeseriesPoint[]) {
+        this.data = data;
```

Comment:
How about multiple charts in this plugin?

Reply:
I had no plans to do that.